icu4c/source/common/putil.cpp

```diff
@@ -1638,34 +1634,25 @@ The leftmost codepage (.xxx) wins.
         return gCorrectedPOSIXLocale;
     }
 
-    if ((p = uprv_strchr(posixID, '.')) != NULL) {
-        /* assume new locale can't be larger than old one? */
-        correctedPOSIXLocale = static_cast<char *>(uprv_malloc(uprv_strlen(posixID)+1));
-        /* Exit on memory allocation error. */
-        if (correctedPOSIXLocale == NULL) {
-            return NULL;
-        }
-        uprv_strncpy(correctedPOSIXLocale, posixID, p-posixID);
-        correctedPOSIXLocale[p-posixID] = 0;
+    // Copy the ID into owned memory.
```

Comment:
FYI I thought I had to add in this function mappings from C/POSIX to en_US_POSIX and started refactoring a bit to make that simpler. Then I discovered that uprv_getPOSIXIDForCategory() already has those mappings, but decided that my changes here still simplify the code, so I left them.